File: problems/27. Remove Element/1 - Brute Force.py
class Solution:
    """
    Approach #1: delete using 'del' operator
    This approach is inefficient because each time then an item will be deleted
    remaining elements will be shifted.

    Time: O(n^2) - the worst case is then all the elements in the list are equal to 'val'
    Space: O(1)
    """

    def removeElement(self, nums: list, val: int) -> int:
        i = 0

        while i < len(nums):
            if nums[i] == val:
                del nums[i]
            else:
                i += 1

        return len(nums)

File: problems/1010. Pairs of Songs With Total Durations Divisible by 60/1 - Brute Force.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        // Time: O(n * n)
        // Space: O(1)

        int n = time.length;
        int res = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if ((time[i] + time[j]) % 60 == 0) res++;
            }
        }

        return res;
    }
}